Python/fileutils.c

```diff
@@ -2450,6 +2473,41 @@ _Py_set_blocking(int fd, int blocking)
     return -1;
 }
 #else   /* MS_WINDOWS */
+int
+_Py_get_blocking(int fd)
+{
+    HANDLE handle;
+    DWORD mode;
+
+    handle = _Py_get_osfhandle(fd);
+    if (handle == INVALID_HANDLE_VALUE)
+        return -1;
+
+    if (!GetNamedPipeHandleState(handle, &mode, NULL, NULL, NULL, NULL, 0)) {
+        PyErr_SetFromWindowsErr(0);
+        return -1;
+    }
+    return mode == PIPE_WAIT;
```

Comment:
The value could include `PIPE_READMODE_MESSAGE` (2) even though it's not something we directly support.

```suggestion
    return mode & PIPE_WAIT;
```